public class ZigzagConversion {
    public static void main(String[] args) {
        String s="safiya";
        int numRows=3;
        System.out.println(convert(s,numRows));
    }
    static String convert(String s,int numRows)
    {
         if(numRows == 1)
        {
            return s;
        }

        StringBuilder[] rows = new StringBuilder[numRows];

        for(int i = 0; i < numRows; i++)
        {
            rows[i] = new StringBuilder();
        }

        int row = 0;
        int direction = 1;

        for(int i = 0; i < s.length(); i++)
        {
            rows[row].append(s.charAt(i));

            if(row == numRows - 1)
            {
                direction = -1;
            }
            else if(row == 0)
            {
                direction = 1;
            }

            row = row + direction;
        }

        StringBuilder result = new StringBuilder();

        for(int i = 0; i < numRows; i++)
        {
            result.append(rows[i]);
        }

        return result.toString();
    }
}
